src/libraries/Microsoft.Extensions.Configuration/src/ChainedConfigurationProvider.cs

```diff
@@ -30,6 +30,11 @@ public ChainedConfigurationProvider(ChainedConfigurationSource source!!)
             _shouldDisposeConfig = source.ShouldDisposeConfiguration;
         }
 
+        /// <summary>
+        /// The chained configuration.
+        /// </summary>
+        public IConfiguration Configuration => _config;
```

Comment:
nit: can we change the constructor implementation from 

```C#
        public ChainedConfigurationProvider(ChainedConfigurationSource source!!)
        {
            if (source.Configuration == null)
            {
                throw new ArgumentException(SR.Format(SR.InvalidNullArgument, "source.Configuration"), nameof(source));
            }
            _config = source.Configuration;
``` 

to 

```C#
        public ChainedConfigurationProvider(ChainedConfigurationSource source!!)
        {
            var configuration = source.Configuration;
            if (configuration  == null)
            {
                throw new ArgumentException(SR.Format(SR.InvalidNullArgument, "source.Configuration"), nameof(source));
            }
            _config = configuration ;
````

The reason `ChainedConfigurationSource .Configuration`  is settable and possible to set it to null I guess between the time of checking and the time assigning it to _config.